registry/storage/filewriter.go

```diff
@@ -22,6 +23,11 @@ type fileWriter struct {
 	err    error // terminal error, if set, reader is closed
 }
 
+type bufferedFileWriter struct {
+	*fileWriter
```

Comment:
Without the pointer, functions appear to be receiving a copy of the embedded struct, not a reference to it, so bufferedFileWriter.offset is never being updated. Instead offset is being updated on a copy of the fileWriter that is being thrown away at the end of freadFromAt.